src/friktion/infos.ts

```diff
@@ -340,4 +354,79 @@ export class VaultInfoWrapper {
   getAPY(): number {
     return this.vaultInfo.snapshotInfo ? this.vaultInfo.snapshotInfo.apy : 0;
   }
+  async getSharePrice(
+    connection: Connection,
+    round?: number,
+    claimDeposit?: boolean,
+    claimWithdrawal?: boolean
+  ): Promise<number> {
+    if (this.vaultInfo.snapshotInfo && this.vaultInfo.roundNumber.toNumber() == round) {
+      return this.vaultInfo.snapshotInfo.shareTokenPrice;
+    } else if (this.vaultInfo.snapshotInfo) {
+      return this.vaultInfo.snapshotInfo.shareTokenPrice;
+    } else if (round) {
+      if (claimWithdrawal) {
+        let roundInfo = this.vaultInfo.roundInfos[round - 1];
+        let underlying = new BN(
+          (
+            await getAccount(connection, this.getRoundUnderlyingTokensPendingWithdrawalsAddress(new BN(round)))
+          ).amount.toString()
+        )
+          .div(new BN(10).pow(new BN(3)))
+          .toNumber();
+        let vaultTokenSupply = roundInfo.voltTokensFromPendingWithdrawals.div(new BN(10).pow(new BN(3))).toNumber();
+        return underlying / vaultTokenSupply;
+      } else if (claimDeposit) {
+        let roundInfo = this.vaultInfo.roundInfos[round - 1];
+        let vaultTokenSupply = new BN(
+          (await getAccount(connection, this.getRoundVoltTokensAddress(new BN(round)))).amount.toString()
+        )
+          .div(new BN(10).pow(new BN(3)))
+          .toNumber();
+        let underlying = roundInfo.underlyingFromPendingDeposits.div(new BN(10).pow(new BN(3))).toNumber();
+        return underlying / vaultTokenSupply;
+      } else {
+        return 1;
+      }
+    } else {
+      round = this.vaultInfo.roundNumber.toNumber() - 1;
+      if (round <= 0) return 1;
+      let roundInfo = this.vaultInfo.roundInfos[round];
+      let mint = this.vaultInfo.vaultMint;
+      let supply = new BN((await getMint(connection, mint)).supply.toString())
+        .add(roundInfo.voltTokensFromPendingWithdrawals)
+        .div(new BN(10).pow(new BN(6)))
+        .toNumber();
+      let totalDeposits = roundInfo.underlyingPreEnter.div(new BN(10).pow(new BN(6))).toNumber();
+      return supply == 0 ? 1 : totalDeposits / supply;
+    }
+  }
+  async getLastTradedOptipon(connection: Connection): Promise<string> {
```

Comment:
- ~~`getLastTradedOptipon`~~ => `getLastTradedOptipon`
- Return object instead of string